web/models/tasks/task.ex

```diff
@@ -21,6 +27,47 @@ defmodule CoursePlanner.Tasks.Task do
     struct
     |> cast(params, @cast_params)
     |> validate_required(@required_params)
-    |> assoc_constraint(:user)
+    |> validate_datetime()
+    |> validate_number(:max_volunteers, greater_than: 0, less_than: 1_000)
   end
+
+  def drop_volunteer(changeset, field, field_data), do: put_assoc(changeset, field, field_data)
+  def update_volunteer(changeset, field, field_data) do
```

Comment:
do we need the `field` parameter? it seems it is always `volunteers` and we already have this information on the function name